R/duplicate_argument_linter.R

```diff
@@ -14,6 +14,11 @@ duplicate_argument_linter <- function(except = character()) {
   xpath_call_with_args <- "//expr[EQ_SUB]"
   xpath_arg_name <- "./EQ_SUB/preceding-sibling::*[1]"
 
+  # Functions that allow sequential updates to variables (#1345)
+  allow_list <- c("mutate", "transmute")
```

Comment:
Instead of this, we can also update default argument for `except` parameter: `except = c("mutate", "transmute")`

I feel this might be better because then users, if they wish to, have the choice to not skip this linter for `dplyr::mutate()` calls.

Reply:
I like the argument solution more. Especially since I remember the repeated updates to cause invalid generated SQL when using dbplyr as a backend in the past, and sequential updates could still reasonably be discouraged in a code style even if they don't error.

Reply:
Sounds good. I have made this change.